fix editphoto

will-moore opened the pull request:
This fixes #255.
EDIT - as discussed, this no-longer fixes 'crop' functionality but removes it:

 - Removed 'crop' functionality (and jQuery imgareaselect plugin)
 - Fixed bug on loading Experimenter Photo if you had other experimenter annotations (it wasn't being filtered by namespace)
 - Fixed 'Delete Photo' functionality. Delete button wasn't being shown before.
 - Added a confirm step so you don't accidentally delete photo.
 - Fixed a couple of layout issues with Change User Photo Submit button etc.

jburel commented:
```
          Traceback (most recent call last):

  File "/home/omero/workspace/OMERO-web/.venv3/lib64/python3.6/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)

  File "/home/omero/workspace/OMERO-web/.venv3/lib64/python3.6/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)

  File "/home/omero/workspace/OMERO-web/.venv3/lib64/python3.6/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)

  File "/home/omero/workspace/OMERO-web/.venv3/lib64/python3.6/site-packages/omeroweb/decorators.py", line 538, in wrapped
    retval = f(request, *args, **kwargs)

  File "/home/omero/workspace/OMERO-web/.venv3/lib64/python3.6/site-packages/omeroweb/decorators.py", line 597, in wrapper
    context = f(request, *args, **kwargs)

  File "/home/omero/workspace/OMERO-web/.venv3/lib64/python3.6/site-packages/omeroweb/webadmin/views.py", line 1106, in manage_avatar
    x1 = int(request.POST.get("x1"))

ValueError: invalid literal for int() with base 10: 'NaN'

<WSGIRequest: POST '/web/webadmin/change_avatar/0/crop/'>
        
```

will-moore commented:
@jburel That commit should fix the error you're seeing, but you don't say how you caused that so I couldn't reproduce/test the fix.

jburel commented:
I clicked on the ``Crop`` button

will-moore commented:
OK, that doesn't cause a problem for me. But if you can try to reproduce, that would be great.

pwalczysko commented:
I cannot reproduce the error @jburel is reporting, but when I tried it, I noticed that the Crop button is "falling out" of the white canvas.

![Screenshot 2021-03-19 at 17 03 39](https://user-images.githubusercontent.com/2478303/111816803-2bd93980-88d5-11eb-914e-9d17903691b9.png)

jburel commented:
Use root and you will see the problem. I reproduced it again without problem

jburel commented:
tested with the root account this am, and with the recent commits it works.

pwalczysko commented:
Yes, I can confirm it works for root too, with clicking of the crop button. 

I think we could do something about the Crop button though https://github.com/ome/omero-web/pull/274#issuecomment-802980304. Further, when a photo of a certain size is uploaded, then the attempt to edit exposes following bug:

The Crop button is completely covered by the image, and it is not possible to click it.
So I guess there are two issues here with the Crop button placement.

will-moore commented:
@pwalczysko looking at the problem, I don't think it's a trivial fix.
We'd have to resize the image, then the crop tool would have to be re-calibrated etc.
I just don't think it's worth investing time into "crop" functionality when anyone can crop an image on their laptop and upload the correct image to start with.
So, unless someone says "we need this", I'll just *remove* all the crop functionality (jQuery plugin and python code) and do any other UI fixes in this PR, instead of fixing the Crop functionality? (This PR wasn't because a user asked for us to fix this).

jburel commented:
I don't think that is a key feature so we should spend too much time on.

pwalczysko commented:
@will-moore @jburel agree with the ultimate removal of the Crop function. I was always wondering why is it there, as the avatar is perfectly fine once I have uploaded the correct image size (and I am not allowed to make any mistake there, which is good). 
Thus, as @jburel says, I would see the deprecation step as a logical one - this might also spare time in the long run, as I remember the Crop as being rpoblematic since several years in quite a few retests

will-moore commented:
Did quite a bit of cleanup:
 - Removed 'crop' functionality (and jQuery imgareaselect plugin)
 - Fixed bug on loading Experimenter Photo if you had other experimenter annotations (it wasn't being filtered by namespace)
 - Fixed 'Delete Photo' functionality. Delete button wasn't being shown before.
 - Added a confirm step so you don't accidentally delete photo.
 - Fixed a couple of layout issues with Change User Photo `Submit` button etc.

snoopycrimecop commented:
Conflicting PR. Removed from build [OMERO-python-superbuild-push#643](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/643/). See the [console output](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/643/consoleText) for more details.
Possible conflicts:
  - PR #180 will-moore 'Update jquery'
    - omeroweb/webadmin/static/webadmin/javascript/jquery.imgareaselect-0.9.10/jquery.imgareaselect.js

--conflicts

will-moore commented:
need to remove failing 'crop' tests: https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-test-integration/lastCompletedBuild/testReport/OmeroWeb.test.integration.test_csrf/TestCsrf/test_avatar/

pwalczysko commented:
Tested today as user-3 on merge-ci.

Testing question
How to achieve ``other experimenter annotations`` ? What type of annotation is exactly meant ? Annotating via CLI ?

 - The Delete button wokrs fine.
 - The crop function is gone
 
RFE: A button saying "Done" or somesuch would be nicer than having to click onto "Back to ...." link in top right. The reflex is to click onto the biggest button in bottom right once done (which uploads a new photo), thus this workflow goes into endless loop.

Bug: The labelling of the upload button seems wrong, see screenslot.

![Screenshot 2021-03-29 at 17 35 42](https://user-images.githubusercontent.com/2478303/112870694-5fc31480-90b6-11eb-8896-bbef7562ba95.png)

pwalczysko commented on omeroweb/webadmin/templates/webadmin/avatar.html:
```diff
@@ -50,7 +47,11 @@ <h1><a href="{% url 'wamyaccount' %}" style="position:relative; top:-38px; float
                 <br />
             {% endfor %}
-             <input type="submit" style="position: absolute; right: 20px; bottom: 20px"{% trans 'Upload' %}" />
+            <input type="submit" style="float:none" value="{% trans 'Upload' %}" />
```
Would it be possible to unify the styles of the "Uploa" and the "Done" buttons ? Atm, the "Done" is appearing with a grey-hue letters, whereas the "Upload" one here with black letters. Maybe simple insertion of ``color: inherit`` into the style definition here would do the trick ?

pwalczysko commented:
One minor comment https://github.com/ome/omero-web/pull/274/commits/74b4161b583bd610ecb54df2299e0dce1d75e707#r607041042 - to be considered for finx only if the situation is indeed as simple as I imaginged in the suggestion.

Otherwise, the whole functionality lgtm. Ready to merge fmpov.

will-moore commented:
Thanks @pwalczysko. I actually went for changing the other button (so they both have darker text) since that's more consistent with other pages.

pwalczysko commented:
Retested on merge-ci, the Upload and Done buttons look the same now, thanks.

Ready to merge fmpov.

will-moore commented:
Please can this be merged. Trying to fix other bugs that are related (https://github.com/ome/omero-web/pull/278#issuecomment-833510312) and didn't realise this wasn't merged yet.
cc @jburel. Thanks.

jburel commented:
Re-tested, things work as expected.
